contracts/kill-switch/KillSwitch.sol

```diff
@@ -0,0 +1,142 @@
+    {
+        blacklistedBaseImplementations[_base] = _denied;
+        emit ChangeBlacklistedBaseImplementation(_base, _denied);
+    }
+
+    function setIssuesRegistry(bytes32 _appId, IIssuesRegistry _issuesRegistry)
+        external
+        authP(CHANGE_ISSUES_REGISTRY_ROLE, arr(_appId, address(appsIssuesSettings[_appId].issuesRegistry), address(_issuesRegistry)))
+    {
+        require(isContract(_issuesRegistry), ERROR_ISSUES_REGISTRY_NOT_CONTRACT);
+        appsIssuesSettings[_appId].issuesRegistry = _issuesRegistry;
+        emit ChangeIssuesRegistry(_appId, address(_issuesRegistry));
+    }
+
+    function setHighestAllowedSeverity(bytes32 _appId, IIssuesRegistry.Severity _severity)
+        external
+        authP(CHANGE_HIGHEST_ALLOWED_SEVERITY_ROLE, arr(_appId, uint256(appsIssuesSettings[_appId].highestAllowedSeverity), uint256(_severity)))
+    {
+        appsIssuesSettings[_appId].highestAllowedSeverity = _severity;
+        emit ChangeHighestAllowedSeverity(_appId, _severity);
+    }
+
+    /**
+     * @dev Note that we are not checking if the appId, base address and instance address are valid and if they correspond
+     *      to each other in order to reduce extra calls. However, since this is only a query method, wrong input
+     *      can only result in invalid output. Internally, this method is used from the Kernel to stop calls if needed,
+     *      and we have several tests to make sure its usage is working as expected.
+     */
+    function shouldDenyCallingApp(bytes32 _appId, address _base, address _instance) external view returns (bool) {
+        // if the instance is whitelisted, then allow given call
+        if (isInstanceWhitelisted(_instance)) {
+            return false;
+        }
+
+        // if the base implementation is blacklisted, then deny given call
+        if (isBaseImplementationBlacklisted(_base)) {
+            return true;
+        }
+
+        // Check if there is a severity issue reported in the corresponding issue registry. If there is actually a
+        // severity issue, check against the kill switch settings if we allow such severity level or not.
+        if (hasExceededAllowedSeverity(_appId, _base)) {
+            return false;
+        }
+
+        // if none of the conditions above were met, then deny given call
+        return true;
+    }
+
+    function isInstanceWhitelisted(address _instance) public view returns (bool) {
+        return whitelistedInstances[_instance];
+    }
+
+    function isBaseImplementationBlacklisted(address _base) public view returns (bool) {
+        return blacklistedBaseImplementations[_base];
+    }
+
+    function hasExceededAllowedSeverity(bytes32 _appId, address _base) public view returns (bool) {
+        IIssuesRegistry.Severity severityFound = getIssuesRegistry(_appId).getSeverityFor(_base);
+        IIssuesRegistry.Severity highestAllowedSeverity = getHighestAllowedSeverity(_appId);
+        return highestAllowedSeverity >= severityFound;
```

Comment:
Nitpick: this should probably be `highestAllowedSeverity < severityFound` (and `shouldDenyCallingApp()` should return true if this function is true), to stay true to this function's name (**exceeding** allowed severity).

Alternatively we could rename this to `isWithinAllowedSeverity()` to keep the logic the same.

Reply:
Good catch, I completely forgot that after renaming it :)